pkg/sql/backfill/backfill.go

```diff
@@ -410,20 +524,38 @@ func (ib *IndexBackfiller) Init(
 		}
 	}
 
-	ib.types = make([]*types.T, len(cols))
-	for i := range cols {
-		ib.types[i] = cols[i].Type
+	return valNeededForCol
+}
+
+// init completes the initialization of an IndexBackfiller.
+func (ib *IndexBackfiller) init(
+	evalCtx *tree.EvalContext,
+	predicateExprs map[sqlbase.IndexID]tree.TypedExpr,
+	valNeededForCol util.FastIntSet,
+	desc *sqlbase.ImmutableTableDescriptor,
+) error {
+	ib.evalCtx = evalCtx
+	ib.predicates = predicateExprs
+
+	// Initialize a list of index descriptors to encode entries for. If there
+	// are no partial indexes, the list is equivalent to the list of indexes
+	// being added. If there are partial indexes, allocate a new list that is
+	// reset in BuildIndexEntriesChunk for every row added.
+	ib.indexesToEncode = ib.added
+	if len(ib.predicates) > 0 {
+		ib.indexesToEncode = make([]sqlbase.IndexDescriptor, 0, len(ib.added))
 	}
-	ib.colIdxMap = make(map[sqlbase.ColumnID]int, len(cols))
-	for i := range cols {
-		ib.colIdxMap[cols[i].ID] = i
+
+	ib.types = make([]*types.T, len(ib.cols))
```

Comment:
Is this used anywhere?

Reply:
Yes here: https://github.com/cockroachdb/cockroach/blob/3b193feb06feac8d4e908a409b1abfad11d57699/pkg/sql/backfill/backfill.go#L481

It's already there on master.